src/python/pants/invalidation/cache_manager.py

```diff
@@ -183,6 +201,11 @@ def create_results_dir(self, root_dir, allow_incremental):
       # If the target is valid, both directories can be assumed to exist.
       return
 
+    # If the vt is invalid, clean. Deletes both because if the stable_dir has somehow been replaced with a real dir,
+    # the relative_symlink call below raises an uncaught exception when it attempts to unlink the real directory.
+    safe_rmtree(current_dir)
+    safe_rmtree(stable_dir)
```

Comment:
I have a followup that does that - I broke it out into a separate commit from due to your original comment. Can add here if you prefer.